client/asset/btc/spv.go

```diff
@@ -922,6 +922,8 @@ func (w *spvWallet) swapConfirmations(txHash *chainhash.Hash, vout uint32, pkScr
 		if assumedMempool {
 			w.log.Tracef("swapConfirmations - scanFilters did not find %v:%d, assuming in mempool.",
 				txHash, vout)
+			// NOT asset.CoinNotFoundError since this is normal for mempool
+			// transactions with an SPV wallet.
```

Comment:
Handling varies in `core`.  This is partly a philosophical question of if it is good to return non-nil errors when nothing unexpected happens.  At least one caller assumed what most callers would assume that a non-nil error is cause for concern.  I drafted the PR this way because it feels like the asset backend is in the position to say if it is cause for concern, not the caller.

Reply:
But the asset backend can't really be sure, e.g. a spv wallet may return no error for a swap output that was not found because it could be in mempool when it's possible that the output really doesn't exist, yet no error is returned. Conversely, a non-spv wallet may return a `CoinNotFoundError` for a coin that does exist and may only subsequently be found when included in a block.

I guess one question is, is it really an error in the semantic sense of the word, for a non-spv wallet to not find a tx output? I think, if it's not an error for spv wallets, it shouldn't be for non-spv wallets either. It's just significantly more reason to worry for non-spv wallets, but is that what the error return here communicates? Or will callers so assume?

I think in terms of communicating that there's reason to worry because a coin was not found, a log should suffice, preferably on the asset backend since only the asset backend knows if there is a cause for worry (i.e. if !spvMode).

Reply:
> I guess one question is, is it really an error in the semantic sense of the word, for a non-spv wallet to not find a tx output? I think, if it's not an error for spv wallets, it shouldn't be for non-spv wallets either.

This depends on two things: (a) spv vs full node, **AND** (b) the caller's expectations.  The second part is important I think because if the caller is looking for confs/spend on **their own swap**, their wallet would certainly be expected to find it even in spv, whereas if it's checking the **counter-party swap**, it's expected not to find it in spv until it's mined.

Reply:
Trying it the other way now, but I'm concerned about big changes since we need to fix these ERR logs in v0.4.